airflow/gcp/operators/campaign_manager.py

```diff
@@ -0,0 +1,312 @@
+
+    def execute(self, context: Dict):
+        hook = GoogleCampaignManagerHook(
+            gcp_conn_id=self.gcp_conn_id,
+            delegate_to=self.delegate_to,
+            api_version=self.api_version,
+        )
+        self.log.info("Deleting Campaign Manager report: %s", self.report_id)
+        hook.delete(profile_id=self.profile_id, report_id=self.report_id)
+        self.log.info("Report deleted.")
+
+
+class GoogleCampaignManagerDownloadReportOperator(BaseOperator):
+    """
+    Retrieves a report and uploads it to GCS bucket.
+
+    .. seealso::
+        Check official API docs:
+        https://developers.google.com/doubleclick-advertisers/v3.3/reports/files/get
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:GoogleCampaignManagerDownloadReportOperator`
+
+    :param profile_id: The DFA user profile ID.
+    :type profile_id: str
+    :param report_id: The ID of the report.
+    :type report_id: str
+    :param file_id: The ID of the report file.
+    :type file_id: str
+    :param bucket_name: The bucket to upload to.
+    :type bucket_name: str
+    :param report_name: The report name to set when uploading the local file.
+    :type report_name: str
+    :param chunk_size: File will be downloaded in chunks of this many bytes.
+    :type chunk_size: int
+    :param api_version: The version of the api that will be requested for example 'v3'.
+    :type api_version: str
+    :param gcp_conn_id: The connection ID to use when fetching connection info.
+    :type gcp_conn_id: str
+    :param delegate_to: The account to impersonate, if any. For this to work, the service accountmaking the
+        request must have  domain-wide delegation enabled.
+    :type delegate_to: str
+    """
+
+    template_fields = ("profile_id", "report_id", "file_id")
+
+    @apply_defaults
+    def __init__(
+        self,
+        profile_id: str,
+        report_id: str,
+        file_id: str,
+        bucket_name: str,
+        report_name: str,
+        chunk_size: int = 5 * 1024 * 1024,
+        api_version: str = "v3.3",
+        gcp_conn_id: str = "google_cloud_default",
+        delegate_to: str = None,
+        *args,
+        **kwargs
+    ):
+        super().__init__(*args, **kwargs)
+        self.profile_id = profile_id
+        self.report_id = report_id
+        self.file_id = file_id
+        self.api_version = api_version
+        self.chunk_size = chunk_size
+        self.bucket_name = self._set_bucket_name(bucket_name)
+        self.report_name = self._set_report_name(report_name)
+        self.gcp_conn_id = gcp_conn_id
+        self.delegate_to = delegate_to
+
+    @staticmethod
+    def _set_report_name(name: str) -> str:
+        return name if name.endswith(".csv") else name + ".csv"
+
+    @staticmethod
+    def _set_bucket_name(name: str) -> str:
+        bucket = name if not name.startswith("gs://") else name[5:]
```

Comment:
I think, we should make _parse_gcs_url publicly and use it here

Reply:
I agree, should it go to utils or to some hook as a static method?